src/Compilers/Core/Portable/InternalUtilities/JsonWriter.cs

```diff
@@ -88,18 +88,56 @@ public void Write(string key, int value)
             Write(value);
         }
 
+        public void Write(string key, int? value)
+        {
+            WriteKey(key);
+            Write(value);
+        }
+
         public void Write(string key, bool value)
         {
             WriteKey(key);
             Write(value);
         }
 
+        public void Write(string key, bool? value)
+        {
+            WriteKey(key);
+            Write(value);
+        }
+
+        public void Write<T>(string key, T value) where T : struct, Enum
+        {
+            WriteKey(key);
+            Write(value.ToString());
```

Comment:
i'm on the fence about this.  i feel like perhaps for enums we just emit the integral value.  the primary concern i have is uncertainty if the runtime always gives the same string value (for things like flags enums) when you call tostring across launches.

Reply:
The `[Flags]` case should be safe as the runtime guarantees the behavior there. From the [doc comments](https://docs.microsoft.com/en-us/dotnet/api/system.enum.tostring?view=net-6.0)

> If the FlagsAttribute is applied and there is a combination of one or more named constants equal to the value of this instance, then the return value is a string containing a delimiter-separated list of the names of the constants.